pkg/kubectl/cmd/create_configmap.go

```diff
@@ -77,18 +79,21 @@ func CreateConfigMap(f *cmdutil.Factory, cmdOut io.Writer, cmd *cobra.Command, a
 		return err
 	}
 	var generator kubectl.StructuredGenerator
+	var resource unversioned.GroupVersionResource
 	switch generatorName := cmdutil.GetFlagString(cmd, "generator"); generatorName {
 	case cmdutil.ConfigMapV1GeneratorName:
 		generator = &kubectl.ConfigMapGeneratorV1{
 			Name:           name,
 			FileSources:    cmdutil.GetFlagStringSlice(cmd, "from-file"),
 			LiteralSources: cmdutil.GetFlagStringSlice(cmd, "from-literal"),
 		}
+		resource = v1.SchemeGroupVersion.WithResource("configmap")
```

Comment:
Agree, though I can see how the current correlation can be misleading. I could see wanting a "v2" generator that exposes new knobs or has different defaults, but still could produce an object that ends up going to a v1 API